### DISCUSSION TITLE
Automatic code splitting but NOT? a bit confusing

### DISCUSSION THREAD

**Discussion by jhonix22** [TanStack/router#2509] [Q&A]
I am new to tanstack router and upon reading the documentation, it says
> 

    [Exceptions to the .lazy.tsx rule](https://tanstack.com/router/latest/docs/framework/react/guide/code-splitting/#exceptions-to-the-lazytsx-rule)

    The __root.tsx route file does not support code splitting, since it's always rendered regardless of the current route.

    ❓ Why can't I keep all of my route code in a single file? Can't you just code split that file?

    It's true, most popular frameworks support automatic code-extraction, but for now, TanStack Router is going to keep things simple by not getting into the business of parsing and extracting code from files. This also allows you to use any bundler you want, via the CLI, without having to worry about whether or not it supports our code extraction approach.

    In practice, we've found that splitting a file into 2 parts (critical and lazy) is the easy part. The harder part by far is wiring it all back together properly using lower-level lazy-loading APIs (not to mention doing that in a type-safe way).

    By keeping the splitting process manual and adhering to a simple set of conventions, there's less room for error and less cognitive overhead when working with code splitting.
    
However, somewhere below it, it also says that you can enable automatic code-splitting via setting `autoCodeSplitting` to **true** inside `tsr.config.json`, with this , you can have one file only and it'll automatically be code-split, is this still true? is this an outdated documentation or am I missing something?

**Comment by schiller-manuel** [comment]
thanks for the reminder, I removed this outdated section.
